addons/im_livechat/static/src/models/partner/partner.js

```diff
@@ -13,11 +15,31 @@ registerClassPatchModel('mail.partner', 'im_livechat/static/src/models/partner/p
     // Public
     //----------------------------------------------------------------------
 
+    convertData(data) {
+        const data2 = this._super(data);
+        if ('livechat_username' in data) {
+            // flux specific, if livechat username is present it means `name`,
+            // `email` and `im_status` have been removed.
+            delete data2.name;
+            delete data2.email;
+            delete data2.im_status;
+            data2.livechat_username = data.livechat_username;
```

Comment:
It's not good to manually remove imperatively remove knowledge. Why no just keep inserting `name`/`email`/`im_status` altogether?

Reply:
Because they contain false even though the value actually exists. It's the only way to avoid leaking the information in stable without changing the API. Cleanup should be done in master.

Reply:
It's much clearer with this explanation.
Could be good to mention that in comment.
It looked strange and the comment is too obscure (they "have been removed"... doesn't really explain why we need to delete them).

Reply:
updated